### PR TITLE
Remov Argmax Computation for torchmetrics in Classification and Segmentation

### PR DESCRIPTION
**Pull request by nilsleh**
Torchmetrics automatically applies the argmax on logits already if the datatype is float (so the logits) for the Classification Metrics https://lightning.ai/docs/torchmetrics/stable/classification/accuracy.html#multiclassaccuracy.

However, some metrics like [Calibration Error](https://lightning.ai/docs/torchmetrics/stable/classification/calibration_error.html#multiclasscalibrationerror) expect the logits. So in order to overwrite the configuration of metrics without having to change the train, val , test steps, this PR removes the argmax computation.

### CODE REVIEW

**Review by calebrob6 — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]
You learn something new every day.
